src/sql/parts/grid/views/query/query.component.ts

```diff
@@ -651,6 +664,43 @@ export class QueryComponent extends GridParentComponent implements OnInit, OnDes
 		}
 	}
 
+	private saveViewState(): void {
+		let gridSelections = this.slickgrids.map(grid => grid.getSelectedRanges());
+		let resultsScrollElement = (this._resultsScrollBox.nativeElement as HTMLElement);
+		let resultsScroll = resultsScrollElement.scrollTop;
+		let messagePaneScroll = (this._messagesContainer.nativeElement as HTMLElement).scrollTop;
+		let slickGridScrolls = this._slickgridElements.map(element => {
+			// Get the slick grid's viewport element and save its scroll position
+			let scrollElement = (element.nativeElement as HTMLElement).children[0].children[3];
```

Comment:
I wonder if this will break easily in the future?  (e.g. if the SlickGrid DOM layout changes around a bit for some reason)

Reply:
It could break if that happens. @anthonydresser is there a better way you know to get the grid's scroll position? I looked but couldn't find anything

Reply:
The slickgrid class has a `getViewport()` function which will return a struct that has the current top row in it. I don't think it's exposed in the angular2-slickgrid component though.

Reply:
Thanks Anthony. Even if I got that info I don't see a way to set it right now without doing similar logic to what I'm doing here. I'll merge it as-is and take a look at improving it in another PR since it sounds like it might involve angular2-slickgrid changes.

Reply:
The grid also has a "scroll row to top function", but yes either way you would need to expose them via the angular 2 component